lib/experimental/new_embed.dart

```diff
@@ -199,6 +200,28 @@ class NewEmbed {
     _initModules().then((_) => _initNewEmbed());
   }
 
+  /// Initializes a listener for messages from the parent window. Allows this
+  /// embedded iframe to display and run arbitrary Dart code.
+  void _initHostListener() {
+    window.addEventListener('message', (dynamic event) {
+      var data = event.data;
+      if (data is! Map) {
+        // Ignore unexpected messages
+        return;
+      }
+
+      var type = data['type'];
+
+      if (type == 'sourceCode') {
```

Comment:
Could there be one for the test code editor as well?

Reply:
I now see that would be a can of worms, since you'd have to add additional code blocks with ids to identify them and so forth. That's something we should still do, but you're smart not to try it in this PR.

Reply:
We could have an additional class that groups code blocks together. Here's an initial draft: https://github.com/dart-lang/dart-pad/issues/1079